### ISSUE TITLE
Move & Resize losing the grip when moving fast

### ISSUE THREAD

**Issue by philip-firstorder** [ailon/markerjs#15]
**Problem:**
When moving or resizing a marker the delta is losing a fraction of a pixel while each `mousemove` event is triggering.
This leads to the mouse cursor not being at the same position as the grip after a while.

 **Solution:**
The solution is to replace the calculation of `dx` & `dy` variables in the `.move()` and `.resize()` functions with some exact computation depending on mouse `event.offsetX` & `event.offsetY`.

I successfully fixed this for the Rectangular and will open a pull request tomorrow after I also fix the line, arrow & text components.

**Comment by philip-firstorder** [comment]
This required more work from me to but I will try to make my pull request today, if not the latest tomorrow. I try to modify as little code as possible.

**Comment by ailon** [comment]
FYI, keep in mind that I've pushed a 1.6 update today. It doesn't touch on anything mouse event related but still.

**Comment by philip-firstorder** [comment]
You implemented previousState, well that is just what I needed, to save the state in my DB and restore it. Thank you very much fo the update. I will get your changes and re-test on the newest version.

**Comment by philip-firstorder** [comment]
Pull request ready, this fixes this ticket. 

I used your **previousState** function to create a protected variable in which I store the state at mousedown.

Then on mousemove I calculate the delta x and y relative to the previousState and move/resize the marker.

I also reset both dragging and resizing variables on the grip mouseup. This fixes a bug where sometimes when I tried to resize after a dragging event, then the marker was also being moved around.

Tested on Chrome and IPhoneX.
